### PR TITLE
Domain fpos

### PR DESCRIPTION
**Pull request by cccs-jh**

### CODE REVIEW

File: src/multidecoder/decoders/network.py [hunk 1]
```diff
@@ -912,6 +943,7 @@ def domain_is_false_positive(domain: bytes) -> bool:
         or re.match(b"[a-z]+[.][A-Z][a-z]+", domain)  # attribute access not domain
         or (len(split) == 3 and split[1] == b"prototype" and len(root) < 3 and len(tld) < 3)  # javascript pattern
         or domain_lower.endswith(b"prototype.at")
+        or b"icrosoft.com".endswith(domain_lower)  # Truncated microsoft.com
```
[the remark this answers is not in the document]
Comment (cccs-jh) [reply src/multidecoder/decoders/network.py:946]: Copilot is missing the edge case. `b"microsoft.com".endswith(domain_lower)` treats `microsoft.com` itself as a false positive, which is why the `m` is removed. A test was added specifically for this case.

File: src/multidecoder/keyword.py [hunk 2]
```diff
@@ -54,9 +54,7 @@ def find_all(keyword: bytes, data: bytes) -> list[int]:
     start = data.find(keyword)
     while start >= 0:
         end = start + len(keyword)
-        if (start == 0 or not data[start - 1 : start].isalnum()) and (
-            end == len(data) or not data[end : end + 1].isalnum()
-        ):
+        if (start == 0 or not chr(data[start - 1]).isalnum()) and (end == len(data) or not chr(data[end]).isalnum()):
```
[the remark this answers is not in the document]
Comment (cccs-jh) [reply src/multidecoder/keyword.py:57]: https://docs.python.org/3.11/library/functions.html#chr

"The valid range for the argument is from 0 through 1,114,111 (0x10FFFF in base 16). [ValueError](https://docs.python.org/3.11/library/exceptions.html#ValueError) will be raised if i is outside that range."

File: tests/test_decoders/test_network.py [hunk 3]
```diff
@@ -197,12 +197,23 @@ def test_DOMAIN_RE_context(data, domain):
         b"view.name",
         b"x.properties",
         b"sub.name",
+        b"oft.com",
```
[the remark this answers is not in the document]
Comment (cccs-jh) [reply tests/test_decoders/test_network.py:200]: `b"oft.com"` is used specifically for the test case as it is marked malicious on [VirusTotal](https://www.virustotal.com/gui/domain/oft.com). Confusing a truncated `microsoft.com` with a known malicious domain is a worse false positive than finding a false positive domain that can be filtered based on registration. `oft.com` is still labelled in contexts where domain identification is reliable, such as within a URL.
